ecommerce/wagtail_views.py

```diff
@@ -0,0 +1,120 @@
+"""Wagtail admin views"""
+from wagtail.admin.views.generic.models import IndexView
+from wagtail.admin.viewsets.base import ViewSetGroup
+from wagtail.admin.viewsets.model import ModelViewSet
+from wagtail.permissions import ModelPermissionPolicy
+
+from ecommerce.models import Product, ProductVersion
+
+
+class CreateOnlyModelPermissionPolicy(ModelPermissionPolicy):
+    """
+    ModelPermissionPolicy to allow `add` permission.
+
+    Wagtail does not support a separate permission to allow read only mode.
+    The reason behind trying to go with a read-only mode is to avoid confusion
+    and allow only a single place to change the price of Products.
+    """
+
+    def user_has_permission(self, user, action):
+        """
+        Allows `add` permission
+        """
+        if action in ("change", "delete"):
+            return False
+        return super().user_has_permission(user, action)
+
+
+class ReadOnlyIndexView(IndexView):
+    """
+    IndexView to override the edit URL with read only inspect URL.
+    """
+
+    def get_edit_url(self, instance):
+        """
+        Returns inspect URL as we have disabled the edit permissions for the Product and ProductVersion.
+        """
+        return self.get_inspect_url(instance)
+
+
+class ProductViewSet(ModelViewSet):
```

Comment:
In General, Should we match the list displays & filters to those of Django Admin - Is It possible?

An example is, The default product queryset provides only active products. That means that in Django admin we can see both the active/inactive products because we override the queryset method and use `Product.all_objects` whereas in Wagtail we see only active ones.

Reply:
It was in my mind to match the List Display and filters, this is the reason that I moved an admin property `get_price` from the admin to models for the Product model. `Price` seemed important information.

Here are the Differences:

- Product List-display has `Is Active` in CMS and `Text ID` in Django admin. `Text ID` is a property type field in the admin and it didn't look like an important one like the price, and it also required to be moved to a common place like models. I can add it if you want.
- Product Version has only 1 difference with the Django admin and that's the ordering of columns. I have changed it.

Reply:
we should if we can.

Reply:
the rest could be done in a follow up if needed

Reply:
Updated the PR to list the in-active products.